### PR TITLE
Update hash_model.rb

### PR DESCRIPTION
**Pull request by garside**
What a long strange trip it's been.

So the gist is:

Between activemodel-6.0 and activemodel-6.1 the underlying attribute writer behavior changed.

Before with `activemodel<=6.0` attribute names could be expressed either as a symbol or string and treated with parity.

Now the `activemodel>=6.1` underlying methods are sensitive to strings vs symbols.

This PR nudges everything in hash model to strings under the hood which its primary consumer (RedisModel) assumes anyway.

I also introduced a `RAILS_VERSION` file to connect together all the gems and allow for easy rolling rails minimum support version (nice plausible reason to keep it) or testing a specific version of rails locally (real reason it exists)

### CODE REVIEW

File: lib/spicerack/hash_model.rb [hunk 1]
```diff
@@ -23,28 +24,44 @@ def inherited(base)
...
+      
+    private
+
+    # The data object can be anything from a normal hash to a hash-like object backed by redis.
+    def synchronize_attribute_from_datastore(name)
+      value_from_datastore = data[name]
+      value_from_attribute = attribute(name)
+      return if value_from_datastore == value_from_attribute
+      
+      value = value_from_datastore || value_from_attribute
```
Comment (yoniamir) [inline lib/spicerack/hash_model.rb:54]: If new value of `value_from_attribute` is nil, dont we want to append it anyway?

File: spec/support/shared_examples/a_hash_model_writer.rb [hunk 2]
```diff
@@ -1,15 +1,15 @@
 # frozen_string_literal: true
 
 RSpec.shared_examples_for "a hash model writer" do |field|
-  subject(:writer) { hash_model.public_send("#{field}=".to_sym, given_value) }
+  subject(:writer) { hash_model.public_send("#{field.to_s}=".to_sym, given_value) }
```
Comment (arettberg) [inline spec/support/shared_examples/a_hash_model_writer.rb:4]: c5
redundo

File: lib/spicerack/hash_model.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@
 require "active_model"
 
 module Spicerack
+  # Proivdes on-demand synchronization between a `data' object and ActiveModel::Attributes
```
Comment (arettberg) [inline lib/spicerack/hash_model.rb:6]: sp

File: lib/spicerack/hash_model.rb (hunk 1 above)
Comment (garside) [reply lib/spicerack/hash_model.rb:54]: naw shouldn't come up. not perceptible here but the code path locks that out by overriding the attribute setter to leverage the data object.

**Review by yoniamir — APPROVED** [review]

**Review by arettberg — APPROVED** [review]

**Pull request merged**